src/string.cr

```diff
@@ -3445,8 +3449,12 @@ class String
   # ```
   def rindex(search : Char, offset = size - 1)
     # If it's ASCII we can delegate to slice
-    if search.ascii? && single_byte_optimizable?
-      return to_slice.rindex(search.ord.to_u8, offset)
+    if single_byte_optimizable?
+      # With `single_byte_optimizable?` there are only ASCII characters and invalid UTF-8 byte
+      # sequences and we can immediately reject any non-ASCII codepoint.
+      return unless search.ascii?
+      
+      return to_slice.fast_rindex(search.ord.to_u8, offset)
```

Comment:
```suggestion
      return to_slice.rindex(search.ord.to_u8, offset)
```

Reply:
Sorry, that was my copy & paste fault. I even remembered to double check the expression for `rindex`, but apparently I was too blind 🙈